worlds/ffmq/Options.py

```diff
@@ -0,0 +1,258 @@
+    Also modify the Desert floor, so it can be navigated without the Mega Grenades and the Dragon Claw."""
+    display_name = "Doom Castle Shortcut"
+
+
+class TweakFrustratingDungeons(Toggle):
+    """Make some small changes to a few of the most annoying dungeons. Ice Pyramid: Add 3 shortcuts on the 1st floor.
+    Giant Tree: Add shortcuts on the 1st and 4th floors and curtail mushrooms population.
+    Pazuzu's Tower: Staircases are devoid of enemies (regardless of Enemies Density settings)."""
+    display_name = "Tweak Frustrating Dungeons"
+
+
+class MapShuffle(Choice):
+    """None: No shuffle. Overworld: Only shuffle the Overworld locations. Dungeons: Only shuffle the dungeons' floors
+    amongst themselves. Temples and Towns aren't included. Overworld And Dungeons: Shuffle the Overworld and dungeons
+    at the same time. Everything: Shuffle the Overworld, dungeons, temples and towns all amongst each others.
+    When dungeons are shuffled, defeating Pazuzu won't teleport you to the 7th floor, you have to get there normally to
+    save the Crystal and get Pazuzu's Chest."""
+    display_name = "Map Shuffle"
+    option_none = 0
+    option_overworld = 1
+    option_dungeons = 2
+    option_overworld_and_dungeons = 3
+    option_everything = 4
+    default = 0
+
+
+class CrestShuffle(Toggle):
+    """Shuffle the Crest tiles amongst themselves."""
+    display_name = "Crest Shuffle"
+
+
+class MapShuffleSeed(FreeText):
+    """If this is a number, it will be used as a set seed number for Map, Crest, and Battlefield Reward shuffles.
+    If this is "random" the seed will be chosen randomly. If it is any other text, it will be used as a seed group name.
+    All players using the same seed group name will get the same shuffle results, as long as their Map Shuffle,
+    Crest Shuffle, and Shuffle Battlefield Rewards settings are the same."""
+    display_name = "Map Shuffle Seed"
+    default = "random"
+
+
+class LevelingCurve(Choice):
+    """Adjust the level gain rate."""
+    display_name = "Leveling Curve"
+    option_half = 0
+    option_normal = 1
+    option_one_and_half = 2
+    option_double = 3
+    option_double_and_half = 4
+    option_triple = 5
+    option_quadruple = 6
+    default = 4
+
+
+class ShuffleBattlefieldRewards(Toggle):
+    """Shuffle the type of reward (Item, XP, GP) given by battlefields and color code them by reward type.
+    Blue: Give an item. Grey: Give XP. Green: Give GP."""
+    display_name = "Shuffle Battlefield Rewards"
+
+
+class BattlefieldsBattlesQuantities(Choice):
+    """Adjust the number of battles that need to be fought to get a battlefield's reward."""
+    display_name = "Battlefields Battles Quantity"
+    option_ten = 0
+    option_seven = 1
+    option_five = 2
+    option_three = 3
+    option_one = 4
+    option_random_one_through_five = 5
+    option_random_one_through_ten = 6
+
+
+option_definitions = {
```

Comment:
You'll want to update your options to a dataclass as seen in the [docs here](https://github.com/ArchipelagoMW/Archipelago/blob/main/docs/options%20api.md#option-creation). 

Access will be simplified in your world as i.e. `self.options.option_name.value` instead of `self.multiworld.option_name[player].value`.